src/Interfaces/AbstractPolyhedron_functions.jl

```diff
@@ -266,42 +266,156 @@ function remove_redundant_constraints(
+                            "invertible matrix"))
+        return false
+    end
+
+    dense_condition = !issparse(M)
+    if !dense_condition
+        throw_error && throw(ArgumentError("the inverse of a sparse matrix can " *
+        "often be dense and can cause the computer to run out of memory. If you " *
+        "are sure you have enough memory, please convert your matrix to a dense matrix; " *
+        "try to pass `Matrix(M)`"))
+        return false
+    end
+    return true
+end
+
+function _check_algorithm_applies(M::AbstractMatrix{N}, P::AbstractPolyhedron{N},
+            ::LinearMapInverseRight; cond_tol=DEFAULT_COND_TOL, throw_error=false) where {N}
+    inv_condition = issquare(M) && isinvertible(M; cond_tol=cond_tol)
+    if !inv_condition
+        throw_error && throw(ArgumentError("algorithm \"inverse_right\" requires an " *
+                            "invertible matrix"))
+        return false
+    end
+    return true
+end
+
+function _check_algorithm_applies(M::AbstractMatrix{N}, P::AbstractPolyhedron{N},
+                                   ::LinearMapLift; throw_error=false) where {N}
+
+    m, n = size(M)
+    size_condition = m > n
+    if !size_condition
+        throw_error && throw(ArgumentError("algorithm \"lift\" requires that the number " *
+            "of rows of the linear map is greater than the number of columns, but " *
+            "they are of size $m and $n respectively"))
+        return false
+    end
+
+    rank_condition = rank(M) == n
+    if !rank_condition
+        throw_error && throw(ArgumentError("the rank of the given matrix is " *
+            "$r, but algorithm \"lift\" requires that it is $n"))
+        return false
+    end
+
+    return true
+end
+
+function _check_algorithm_applies(M::AbstractMatrix{N}, P::AbstractPolyhedron{N},
+                                   ::LinearMapVRep; throw_error=false) where {N}
+
+    # TODO: the second check should be !isbounded(P) but this may be expensive;
+    # see also #998
+    if !applicable(vertices_list, P) || (P isa HPolyhedron)
+        throw_error && throw(ArgumentError("algorithm \"vrep\" requires that the number " *
+            "list of vertices of the polyhedron is applicable, but it is not for a polyhedron " *
+            "of type $(typeof(P))"))
+        return false
+    end
+    return true
+end
+
+function _default_linear_map_algorithm(M::AbstractMatrix{N}, P::AbstractPolyhedron{N};
+                cond_tol=DEFAULT_COND_TOL, backend=nothing, elimination_method=nothing) where {N}
+
+    if _check_algorithm_applies(M, P, LinearMapInverse(), cond_tol=cond_tol)
+        algo = LinearMapInverse(inv(M))
+    elseif _check_algorithm_applies(M, P, LinearMapLift())
+        algo = LinearMapLift()
+    else
+        require(:Polyhedra; fun_name="linear_map with elimination")
+        require(:CDDLib; fun_name="linear_map with elimination")
+        if backend == nothing
+            backend = default_cddlib_backend(N)
+        end
```

Comment:
I think it's fine to keep the `require` inside the clause. Or i could move these require to the actual algorithm `_linear_map_hrep` (as it is the case for the `vrep` algorithm).

The fact of requiring `CDDLib` is that it is the only library (afaik) that does variable elimination.

Reply:
> The fact of requiring `CDDLib` is that it is the only library (afaik) that does variable elimination.

Fine, but it should still make sense to move it inside the `if` because at some point there may be a different package.

Reply:
Ok, i see what you mean now :+1: 

```julia
        require(:Polyhedra; fun_name="linear_map with elimination")
        if backend == nothing
            require(:CDDLib; fun_name="linear_map with elimination")
            backend = default_cddlib_backend(N)
        end
```